camel/agents/chat_agent.py

```diff
@@ -91,20 +117,23 @@ class ChatAgent(BaseAgent):
             messages to include in the context window. If `None`, no windowing
             is performed. (default: :obj:`None`)
         output_language (str, optional): The language to be output by the
-        agent. (default: :obj:`None`)
+            agent. (default: :obj:`None`)
+        function_list (Optional[List[OpenAIFunction]]): List of OpenAIFunction
+            objects (default: :obj:`None`)
     """
 
     def __init__(
         self,
         system_message: BaseMessage,
         model: Optional[ModelType] = None,
-        model_config: Optional[Any] = None,
+        model_config: Optional[BaseConfig] = None,
         message_window_size: Optional[int] = None,
         output_language: Optional[str] = None,
+        function_list: Optional[List[OpenAIFunction]] = None,
     ) -> None:
 
         self.orig_sys_message: BaseMessage = system_message
-        self.system_message: BaseMessage = system_message
+        self.system_message = system_message
```

Comment:
Actually the way it is implemented is nice. instead of the direct assignment to the frivate field, it goes through the property setter. I suggest not to make the change that you @lightaime propose. It works fine, I had checked it.

Reply:
Oh, yeah, you are right. Please ignore this comment.